ros_buildfarm/common.py

```diff
@@ -444,3 +444,14 @@ def topological_order_packages(packages):
         if pkg_path is None:
             raise RuntimeError('Circular dependency in: %s' % pkg)
     return ordered_pkg_tuples
+
+
+def get_node_label(
+    config_job_label, rosdistro_name=None, job_type=None, build_name=None
+):
+    if config_job_label is not None:
+        return config_job_label
+    label = 'buildslave'
+    if rosdistro_name and job_type and build_name:
```

Comment:
Only when all the arguments are provided for the second label it is being added.